<!--REQUEST_ID=eafa3533-748d-4907-b880-4f8f5ae0b48a-->
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48021&attempt_id=a8c098f4-700d-49d1-be2b-d615390604e1)

Attempted: uhub

<details><summary>Partial log (click to expand)</summary><p>

```
                 from /build/uhub-0.5.0/src/uhub.h:28,
                 from /build/uhub-0.5.0/autotest/test_commands.tcc:1,
                 from /build/uhub-0.5.0/autotest/test.c:47:
/nix/store/sfn26s5v25ij5dpm25wai61kzn1qib3d-glibc-2.27-dev/include/string.h:136:12: note: in a call to function 'strcmp' declared here
 extern int strcmp (const char *__s1, const char *__s2)
            ^~~~~~
make[1]: *** [CMakeFiles/Makefile2:224: CMakeFiles/test.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
builder for '/nix/store/3qg30r1n0spprdvipnm8si0grd7v2wca-uhub-0.5.0.drv' failed with exit code 2
error: build of '/nix/store/3qg30r1n0spprdvipnm8si0grd7v2wca-uhub-0.5.0.drv' failed
```
</p></details>

--- a/pkgs/servers/uhub/default.nix
+++ b/pkgs/servers/uhub/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, fetchFromGitHub, cmake, openssl, sqlite, pkgconfig, systemd
+{ stdenv, fetchpatch, fetchFromGitHub, cmake, openssl, sqlite, pkgconfig, systemd
 , tlsSupport ? false }:
 
 assert tlsSupport -> openssl != null;
@@ -31,9 +31,14 @@ stdenv.mkDerivation rec {
 
   patches = [
     ./plugin-dir.patch
+    # fix aarch64 build: https://github.com/janvidar/uhub/issues/46
+    (fetchpatch {
+      url = "https://github.com/janvidar/uhub/pull/47.patch";
+      sha256 = "07yik6za89ar5bxm7m2183i7f6hfbawbxvd4vs02n1zr2fgfxmiq";
+    })
 
     # Fixed compilation on systemd > 210
-    (fetchurl {
+    (fetchpatch {
       url = "https://github.com/janvidar/uhub/commit/70f2a43f676cdda5961950a8d9a21e12d34993f8.diff";
       sha256 = "1jp8fvw6f9jh0sdjml9mahkk6p6b96p6rzg2y601mnnbcdj8y8xp";
     })

--- a/pkgs/servers/uhub/default.nix
+++ b/pkgs/servers/uhub/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, cmake, openssl, sqlite, pkgconfig, systemd
+{ stdenv, fetchurl, fetchFromGitHub, cmake, openssl, sqlite, pkgconfig, systemd
 , tlsSupport ? false }:
 
 assert tlsSupport -> openssl != null;
@@ -7,9 +7,11 @@ stdenv.mkDerivation rec {
   name = "uhub-${version}";
   version = "0.5.0";
 
-  src = fetchurl {
-    url = "https://www.extatic.org/downloads/uhub/uhub-${version}-src.tar.bz2";
-    sha256 = "1xcqjz20lxikzn96f4f69mqyl9y985h9g0gyc9f7ckj18q22b5j5";
+  src = fetchFromGitHub {
+    owner = "janvidar";
+    repo = "uhub";
+    rev = version;
+    sha256 = "0zdbxfvw7apmfhqgsfkfp4pn9iflzwdn0zwvzymm5inswfc00pxg";
   };
 
   nativeBuildInputs = [ pkgconfig ];

